pkg/jx/cmd/delete_app.go

```diff
@@ -90,23 +93,82 @@ func NewCmdDeleteApp(f Factory, in terminal.FileReader, out terminal.FileWriter,
 	cmd.Flags().StringVarP(&options.SelectFilter, "filter", "f", "", "Filter the list of apps to those containing this text")
 	cmd.Flags().StringVarP(&options.Timeout, optionTimeout, "t", "1h", "The timeout to wait for the promotion to succeed in the underlying Environment. The command fails if the timeout is exceeded or the promotion does not complete")
 	cmd.Flags().StringVarP(&options.PullRequestPollTime, optionPullRequestPollTime, "", "20s", "Poll time when waiting for a Pull Request to merge")
+	// TODO - Create an Application CRD that gets populated with the org when an application is created/imported to store this.
+	cmd.Flags().StringVarP(&options.Org, "org", "o", "", "github organisation/project name that source code resides in. Temporary workaround until the platform can determine this automatically")
 	cmd.Flags().BoolVarP(&options.BatchMode, "batch-mode", "b", false, "Run without being prompted. WARNING! You will not be asked to confirm deletions if you use this flag.")
 
 	return cmd
 }
 
 // Run implements this command
 func (o *DeleteAppOptions) Run() error {
+	err := o.init()
+	if err != nil {
+		return err
+	}
+
+	isProw, err := o.isProw()
+	if err != nil {
+		return err
+	}
+
+	var deletedApplications []string
+	if isProw {
+		deletedApplications, err = o.deleteProwApp()
+	} else {
+		deletedApplications, err = o.deleteJenkinsApp()
+	}
+
+	if err != nil {
+		return errors.Wrapf(err, "deleting application")
+	}
+	log.Infof("Deleted Application(s): %s\n", util.ColorInfo(strings.Join(deletedApplications, ",")))
+	return nil
+}
+
+func (o *DeleteAppOptions) deleteProwApp() (deletedApplications []string, err error) {
+	if o.Org == "" {
+		return deletedApplications, errors.New("--org must be supplied")
```

Comment:
should this not return an empty array?  (ie if you ask to delete 5 apps and one failed I would expect a return with the 4 successfully deleted ones and an error?

(at least it implies the return is the applications that have been deleted)

Reply:
it does return an empty array. Due to the magic of go, the `deletedApplications` that is defined as a named return value gets initialised to the `nil` value at the start of the function. The `nil` value of an array/slice in go is (very much to my liking) an empty array/slice